lib/vm/src/instance/mod.rs

```diff
@@ -1037,6 +1037,27 @@ impl InstanceHandle {
         Ok(())
     }
 
+    /// Similar to finish_instantiation, except it doesn't invoke start function.
+    pub unsafe fn finish_instantiation_without_start_func(
```

Comment:
Can we avoid copy-pasing every function and instead use delegation? 

Ie, rather making `finish_instantiation_without_start_func` a copy of `finish_instantiation`, can we write

```rust 
    pub unsafe fn finish_instantiation(
        &self,
        trap_handler: &dyn TrapHandler,
        data_initializers: &[DataInitializer<'_>],
    ) -> Result<(), Trap> {
      self.finish_instantiation_without_start_func(data_initializers);
      self.call_start_function(trap_handler)?;
      Ok(())
    }
```

? If we can, we probably should do this pattenrn for all functions duplicated so far. 

I feel this is the code we might leave with for a long time, so it's important to keep it clean.